smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -194,68 +95,122 @@ def __init__(
-        """Ensure the request can be processed.
-        :param request: The request to validate
-        :return: True if the request is valid, False otherwise"""
-        if not self._feature_store:
-            if request.model_key:
-                logger.error("Unable to load model by key without feature store")
-                return False
-
-            if request.input_keys:
-                logger.error("Unable to load inputs by key without feature store")
-                return False
-
-            if request.output_keys:
-                logger.error("Unable to persist outputs by key without feature store")
-                return False
-
-        if not request.model_key and not request.raw_model:
-            logger.error("Unable to continue without model bytes or feature store key")
-            return False
+    def _check_feature_stores(self, request: InferenceRequest) -> bool:
+        """Ensures that all feature stores required by the request are available
 
-        if not request.input_keys and not request.raw_inputs:
-            logger.error("Unable to continue without input bytes or feature store keys")
+        :param request: The request to validate
+        :returns: False if feature store validation fails for the request, True otherwise
+        """
+        # collect all feature stores required by the request
+        fs_model: t.Set[str] = set()
+        if request.model_key:
+            fs_model = {request.model_key.descriptor}
+        fs_inputs = {key.descriptor for key in request.input_keys}
+        fs_outputs = {key.descriptor for key in request.output_keys}
+
+        # identify which feature stores are requested and unknown
+        fs_desired = fs_model.union(fs_inputs).union(fs_outputs)
+        fs_actual = {item.descriptor for item in self._feature_stores.values()}
+        fs_missing = fs_desired - fs_actual
+
+        if self._fs_factory is None:
+            logger.warning("No feature store factory configured")
             return False
 
-        if request.callback is None:
-            logger.error("No callback channel provided in request")
-            return False
+        # create the feature stores we need to service request
+        if fs_missing:
+            logger.debug(f"Adding feature store(s): {fs_missing}")
+            for descriptor in fs_missing:
+                feature_store = self._fs_factory(descriptor)
+                self._feature_stores[descriptor] = feature_store
 
         return True
 
+    def _check_model(self, request: InferenceRequest) -> bool:
+        """Ensure that a model is available for the request
+
+        :param request: The request to validate
+        :returns: False if model validation fails for the request, True otherwise
+        """
+        if request.model_key or request.raw_model:
+            return True
+
+        logger.error("Unable to continue without model bytes or feature store key")
```

Comment:
This used ``logger.error`` but ``_check_feature_stores()`` used ``logger.warning`` (line 149).  Should they use the same?

Reply:
My thought here was "if the request is bad, it can never be serviced but if the component hasn't yet been loaded, it still could be"... but I think that may not really be much of a difference in practical terms.

Replaced `warning` with `error`